x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/action/TransportGetStackTracesAction.java

```diff
@@ -371,33 +403,49 @@ private void searchEventGroupedByStackTrace(
                 long totalFinalCount = 0;
                 Map<TraceEventID, TraceEvent> stackTraceEvents = new HashMap<>(MAX_TRACE_EVENTS_RESULT_SIZE);
 
-                Terms executableNames = searchResponse.getAggregations().get("group_by");
-                for (Terms.Bucket executableBucket : executableNames.getBuckets()) {
-                    String executableName = executableBucket.getKeyAsString();
-
-                    Terms threads = executableBucket.getAggregations().get("group_by");
-                    for (Terms.Bucket threadBucket : threads.getBuckets()) {
-                        String threadName = threadBucket.getKeyAsString();
-
-                        Terms hosts = threadBucket.getAggregations().get("group_by");
-                        for (Terms.Bucket hostBucket : hosts.getBuckets()) {
-                            String hostID = hostBucket.getKeyAsString();
-
-                            Terms stacktraces = hostBucket.getAggregations().get("group_by");
-                            for (Terms.Bucket stacktraceBucket : stacktraces.getBuckets()) {
-                                Sum count = stacktraceBucket.getAggregations().get("count");
-                                int finalCount = resampler.adjustSampleCount((int) count.value());
-                                if (finalCount <= 0) {
-                                    continue;
+                for (Terms.Bucket samplingFrequencyBucket : samplingFrequencies.getBuckets()) {
+                    log.debug(
+                        "Using sampling frequency [{}] for [{}] stacktrace events.",
+                        samplingFrequencyBucket.getKeyAsString(),
+                        totalCount
```

Comment:
Isn't this debug log message misleading as we're using `totalCount` (which spans all frequency buckets)

Reply:
Removed the debug log, was only needed for debugging during development.